src/api/route-services/pipeline-response.js

```diff
@@ -53,7 +54,26 @@ const pipelineResponse = async (io, message) => {
     Promise.all(plotConfigUploads.map((p) => p.catch((e) => e)));
   }
 
-  experimentService.updateProcessingConfig(experimentId, [{ name: taskName, body: output.config }]);
+  const {
+    processingConfig: currentConfig,
+  } = await experimentService.getProcessingConfig(experimentId);
+
+  if (sampleUuid !== '') {
+    await experimentService.updateProcessingConfig(experimentId, [
+      {
+        name: taskName,
+        body: { ...currentConfig[taskName], [sampleUuid]: output.config },
+      },
+    ]);
+  } else {
+    await experimentService.updateProcessingConfig(experimentId, [
+      {
+        name: taskName,
+        body: output.config,
+      },
+    ]);
+  }
+
```

Comment:
If the task is not per-sample, just overwrite the current config as we did previously. If it is, create or update the sample-specific config with the result.